presto-iceberg/src/main/java/com/facebook/presto/iceberg/optimizer/IcebergFilterPushdown.java

```diff
@@ -0,0 +1,258 @@
+
+    public IcebergFilterPushdown(IcebergTransactionManager transactionManager,
+                                 RowExpressionService rowExpressionService,
+                                 StandardFunctionResolution functionResolution,
+                                 FunctionMetadataManager functionMetadataManager,
+                                 IcebergResourceFactory resourceFactory,
+                                 HdfsEnvironment hdfsEnvironment,
+                                 TypeManager typeManager)
+    {
+        this.icebergTransactionManager = requireNonNull(transactionManager, "transactionManager is null");
+        this.rowExpressionService = requireNonNull(rowExpressionService, "rowExpressionService is null");
+        this.functionResolution = requireNonNull(functionResolution, "functionResolution is null");
+        this.functionMetadataManager = requireNonNull(functionMetadataManager, "functionMetadataManager is null");
+        this.resourceFactory = requireNonNull(resourceFactory, "resourceFactory is null");
+        this.hdfsEnvironment = requireNonNull(hdfsEnvironment, "hdfsEnvironment is null");
+        this.typeManager = requireNonNull(typeManager, "typeManager is null");
+    }
+
+    @Override
+    public PlanNode optimize(PlanNode maxSubplan, ConnectorSession session, VariableAllocator variableAllocator, PlanNodeIdAllocator idAllocator)
+    {
+        if (!isPushdownFilterEnabled(session)) {
+            return maxSubplan;
+        }
+        return rewriteWith(new IcebergFilterPushdownRewriter(session, idAllocator), maxSubplan);
+    }
+
+    protected ConnectorPushdownFilterResult pushdownFilter(
+            ConnectorSession session,
+            IcebergAbstractMetadata metadata,
+            ConnectorTableHandle tableHandle,
+            ConnectorTransactionHandle transactionHandle,
+            RowExpression filter,
+            Optional<ConnectorTableLayoutHandle> currentLayoutHandle)
+    {
+        Result result = checkConstantBooleanExpression(rowExpressionService, functionResolution, filter, currentLayoutHandle, metadata, session);
+        if (result.getConnectorPushdownFilterResult() != null) {
+            return result.getConnectorPushdownFilterResult();
+        }
+
+        Map<String, ColumnHandle> columnHandles = metadata.getColumnHandles(session, tableHandle);
+        TupleDomain<ColumnHandle> entireColumnDomain = getEntireColumnDomain(result, columnHandles);
+
+        Constraint<ColumnHandle> constraint = new Constraint<>(entireColumnDomain);
+
+        // Extract deterministic conjuncts that apply to partition columns and specify these as Constraint#predicate
+        constraint = extractDeterministicConjuncts(
+                rowExpressionService,
+                functionResolution,
+                functionMetadataManager,
+                session,
+                result,
+                columnHandles,
+                entireColumnDomain,
+                constraint);
```

Comment:
THis constraint parameter is not used. Shall we remove it?

Reply:
The reason I included it in the method call was to return the original constraint in case none of the if conditions are satisfied in the `extractDeterministicConjuncts` method. Let me know if we should not send the original constraint and add that check in the source method (`com.facebook.presto.iceberg.optimizer.IcebergFilterPushdown#pushdownFilter`)